src/App.tsx

```diff
@@ -10,33 +12,22 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTabId, setSelectedTab] = useState(tabs[0].id);
+
+  const onTabSelected = (tab: Tab) => setSelectedTab(tab.id);
+
+  const selectedTab = tabs.find(tab => tab.id === selectedTabId) || tabs[0];
```

Comment:
you duplicate tab finding logic twice  - fix it (you may store tab object for example )